financespy/memory_backend.py

```diff
@@ -0,0 +1,21 @@
+import collections
+from financespy.transaction import parse_transaction
+from financespy.transaction import Transaction
+
+class MemoryBackend:
+    def __init__(self):
+        self._months = collections.defaultdict(
+            lambda: [ [] for i in range(0, 31)]
+        )
+
+    def insert_record(self, date, record):
+        if type(record) is str:
```

Comment:
```suggestion
        if type(record) is not Transaction:
            raise TypeError("Supplied parameter is not a transaction")
        if type(record) is str:
            record = parse_transaction(record)
```

Reply:
Maybe the test for "is str" should come first, otherwise it will never be reached if you supply a string to be parsed instead of a Transaction object. Do you agree? This scheme is needed since Python doesn't support method overloading.

Reply:
I'll check this and let you know, and add a unit test as well.